src/test/unit/services/sample/standalone_gqs_multidim_test.cpp

```
@@ -0,0 +1,81 @@
+ }
+
+  void TearDown() {
+    delete(model);
+  }
+
+  stan::test::unit::instrumented_interrupt interrupt;
+  std::stringstream logger_ss;
+  stan::callbacks::stream_logger logger;
+  stan_model *model;
+};
+
+TEST_F(ServicesStandaloneGQ2, genDraws_gq_test_multidim) {
+  stan::io::stan_csv multidim_csv;
+  std::stringstream out;
+  std::ifstream csv_stream;
+  csv_stream.open("src/test/test-models/good/services/gq_test_multidim_fit.csv");
+  multidim_csv = stan::io::stan_csv_reader::parse(csv_stream, &out);
+  csv_stream.close();
+  EXPECT_EQ(12345U, multidim_csv.metadata.seed);
+  ASSERT_EQ(247, multidim_csv.header.size());
+  EXPECT_EQ("p_ar_mat[1,1,1,1]", multidim_csv.header(7));
+  EXPECT_EQ("p_ar_mat[4,5,2,3]", multidim_csv.header(126));
+  EXPECT_EQ("gq_ar_mat[1,1,1,1]", multidim_csv.header(127));
+  EXPECT_EQ("gq_ar_mat[4,5,2,3]", multidim_csv.header(246));
+  ASSERT_EQ(1000, multidim_csv.samples.rows());
+  ASSERT_EQ(247, multidim_csv.samples.cols());
+
+  // model gq_test_multidim has 1 param, length 120
+  std::vector<std::string> param_names;
+  std::vector<std::vector<size_t>> param_dimss;
+  stan::services::get_model_parameters(*model, param_names, param_dimss);
+
+  EXPECT_EQ(param_names.size(), 1);
+  EXPECT_EQ(param_dimss.size(), 1);
+  int total = 1;
+  for (size_t i = 0; i < param_dimss[0].size(); ++i) {
+    total *= param_dimss[0][i];
+  }
+  EXPECT_EQ(total, 120);
+
+  std::stringstream sample_ss;
+  stan::callbacks::stream_writer sample_writer(sample_ss, "");
+  int return_code = stan::services::standalone_generate(*model,
+                                    multidim_csv.samples.middleCols<120>(7),
+                                    12345,
+                                    interrupt,
+                                    logger,
+                                    sample_writer);
+  if (return_code != stan::services::error_codes::OK)
+    std::cout << "ERROR: " << logger_ss.str() << std::endl;
```

This needs to actually assert failure rather than just print a string

---

that test is there on line 78 - the test will fail.
it was useful during development - if the test fails, you get immediate feedback on why.
if this is a problem for Travis or other CI tools, I can remove it.

---

Ah, I see. I would have expected the testing framework to provide the exact same service...

---

Probably best to just take them out and use the testing framework to figure out where test failures happen.

---

will do.